OSDOCS-2917: Azure Ephemeral OS disk support

jeana-redhat opened the pull request:
For [OSDOCS-2917](https://issues.redhat.com/browse/OSDOCS-2917)

Will also need release notes announcement in the enterprise-4.10 branch.

Preview: new sections in _Creating a machine set on Azure_

- [Machine sets that deploy machines on Ephemeral OS disks](https://deploy-preview-40365--osdocs.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-azure.html#machineset-azure-ephemeral-os_creating-machineset-azure)
- [Creating machines on Ephemeral OS disks by using machine sets](https://deploy-preview-40365--osdocs.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-azure.html#machineset-creating-azure-ephemeral-os_creating-machineset-azure)

JoelSpeed reviewed (commented):
LGTM

jeana-redhat commented:
@huali9 could you please verify this procedure for the user docs when you have time? It's for https://issues.redhat.com/browse/OCPCLOUD-1303

jeana-redhat commented:
Thanks, @huali9 :slightly_smiling_face: I do not see your two comments, I hope GitHub did not eat them :grimacing: Can you try reposting please?

huali9 commented:
> Thanks, @huali9 🙂 I do not see your two comments, I hope GitHub did not eat them 😬 Can you try reposting please?

Sorry, @jeana-redhat Can you see the comments now?

jeana-redhat commented:
Yes, I see them now! Thanks Huali!

jeana-redhat commented:
Update: Addressed @huali9's comments (thanks!) and moved the placement type warning to the instructions since it's related to the `providerSpec` field edits.

mburke5678 commented on modules/machineset-azure-ephemeral-os.adoc:
```diff
@@ -0,0 +1,12 @@
+[id="machineset-azure-ephemeral-os_{context}"]
+= Machine sets that deploy machines on Ephemeral OS disks
+
+You can save on costs by creating a machine set running on Azure that deploys machines on Ephemeral OS disks. Ephemeral OS disks use local VM capacity rather than remote Azure Storage. This configuration therefore incurs no additional cost and provides lower latency for reading, writing, and reimaging.
```
It seems a bit odd to lead with cost savings. Maybe that's just me. 
_You can create a machine set running on Azure..._

mburke5678 commented on modules/machineset-creating-azure-ephemeral-os.adoc:
```diff
@@ -0,0 +1,46 @@
+
+.Procedure
+
+. Add the following to the `providerSpec` field:
```
Should we put a step to edit the machine set YAML to the procedure, in case folks skip the intro paragraph? 
Is there a specific machine set or whichever  one moves the nodes you want on Ephemeral OS disks?

mburke5678 commented:
@jeana-redhat A couple of minor nits. Otherwise LGTM

huali9 commented:
LGTM

jeana-redhat commented:
/cherrypick enterprise-4.10
